src/main/java/de/rwth/idsg/steve/service/CentralSystemService16_Service.java

```diff
@@ -276,6 +289,117 @@ public DataTransferResponse dataTransfer(DataTransferRequest parameters, String
         return new DataTransferResponse().withStatus(DataTransferStatus.ACCEPTED);
     }
 
+    public SignCertificateResponse signCertificate(SignCertificate parameters, String chargeBoxIdentity) {
+        log.info("Received SignCertificateRequest from '{}' with CSR length: {}", chargeBoxIdentity,
+                parameters.getCsr() != null ? parameters.getCsr().length() : 0);
+
+        var response = new SignCertificateResponse();
+
+        try {
+            var csr = parameters.getCsr();
+            if (csr == null || csr.trim().isEmpty()) {
+                log.error("Empty or null CSR received from '{}'", chargeBoxIdentity);
+                response.setStatus(SignCertificateResponse.GenericStatusEnumType.REJECTED);
+                return response;
+            }
+
+            if (!certificateSigningService.isInitialized()) {
+                log.error("Certificate signing service not initialized. Check TLS configuration.");
+                response.setStatus(SignCertificateResponse.GenericStatusEnumType.REJECTED);
+
+                return response;
+            }
+
+            var signedCertificatePem = certificateSigningService.signCertificateRequest(csr, chargeBoxIdentity);
+            var caCertificatePem = certificateSigningService.getCertificateChain();
+            var certificateChain = signedCertificatePem + caCertificatePem;
```

Comment:
Could be grouped into a single method in the service that returns the signedCertificatePem and the certificateChain length